feature/interface/singleton/otg_tests/singleton_test/singleton_test.go

```diff
@@ -246,6 +276,64 @@ func (tc *testCase) verifyATE(t *testing.T) {
 	})
 }
 
+func (tc *testCase) configureIPv4FlowHeader(t *testing.T, packetSize uint16) {
+	otg := tc.ate.OTG()
+	flow := tc.top.Flows().Items()[0]
+	flow.TxRx().Device().SetTxNames([]string{ateSrc.Name + ".IPv4"}).SetRxNames([]string{ateDst.Name + ".IPv4"})
+	flow.Size().SetFixed(int32(packetSize))
+	v4 := flow.Packet().Add().Ipv4()
+	v4.Src().SetValue(ateSrc.IPv4)
+	v4.Dst().SetValue(ateDst.IPv4)
+	tc.ate.OTG().PushConfig(t, tc.top)
+	otg.Telemetry().Interface(ateSrc.Name+".Eth").Ipv4NeighborAny().LinkLayerAddress().Watch(
+		t, time.Minute, func(val *otgtelemetry.QualifiedString) bool {
+			return val.IsPresent()
+		}).Await(t)
+}
+
+func (tc *testCase) configureIPv4DfFlowHeader(t *testing.T, packetSize uint16) {
+	otg := tc.ate.OTG()
+	flow := tc.top.Flows().Items()[0]
+	flow.TxRx().Device().SetTxNames([]string{ateSrc.Name + ".IPv4"}).SetRxNames([]string{ateDst.Name + ".IPv4"})
+	v4 := flow.Packet().Add().Ipv4()
+	v4.DontFragment().SetValue(1)
+	v4.Src().SetValue(ateSrc.IPv4)
+	v4.Dst().SetValue(ateDst.IPv4)
+	tc.ate.OTG().PushConfig(t, tc.top)
+	otg.Telemetry().Interface(ateSrc.Name+".Eth").Ipv4NeighborAny().LinkLayerAddress().Watch(
+		t, time.Minute, func(val *otgtelemetry.QualifiedString) bool {
+			return val.IsPresent()
+		}).Await(t)
+}
+
+func (tc *testCase) configureIPv6FlowHeader(t *testing.T, packetSize uint16) {
+	otg := tc.ate.OTG()
+	flow := tc.top.Flows().Items()[0]
+	flow.TxRx().Device().SetTxNames([]string{ateSrc.Name + ".IPv6"}).SetRxNames([]string{ateDst.Name + ".IPv6"})
+	v6 := flow.Packet().Add().Ipv6()
+	v6.Src().SetValue(ateSrc.IPv6)
+	v6.Dst().SetValue(ateDst.IPv6)
+	tc.ate.OTG().PushConfig(t, tc.top)
+	otg.Telemetry().Interface(ateSrc.Name+".Eth").Ipv6NeighborAny().LinkLayerAddress().Watch(
+		t, time.Minute, func(val *otgtelemetry.QualifiedString) bool {
+			return val.IsPresent()
+		}).Await(t)
+}
+
+func (tc *testCase) waitOTGIPv4ARPEntry(t *testing.T) {
+	tc.ate.OTG().Telemetry().Interface(ateSrc.Name+".Eth").Ipv4NeighborAny().LinkLayerAddress().Watch(
+		t, time.Minute, func(val *otgtelemetry.QualifiedString) bool {
+			return val.IsPresent()
+		}).Await(t)
+}
+
+func (tc *testCase) waitOTGIPv6ARPEntry(t *testing.T) {
```

Comment:
ND is the more accurate term for IPv6, not ARP. Or change both IPv4 and IPv6 ARP to just Neighbor?

Reply:
Replaced with "Neighbor"